src/dsp_tools/xmllib/models/dsp_base_resources.py

```diff
@@ -50,11 +42,15 @@ def add_comments(self, comments: list[str]) -> AnnotationResource:
         return self
 
     def serialise(self) -> etree._Element:
+        self._check_for_and_convert_unexpected_input()
```

Comment:
Are you sure you want to move these checks from the post_init to the serialise method? To me, validation should be done right away, not in the very end.

Reply:
I agree conceptually. But we allow people to add more information later. That will not be checked during post init (correct me if I am wrong). So we may not notice some mistakes unless we check at the very end.

Reply:
Good point. I suppose the "clean" way would then be to also check in the methods that add this information. But maybe it's just simpler to go for checking in the end.

Reply:
Sounds reasonable